pandas/core/base.py

```diff
@@ -926,14 +926,14 @@ def to_numpy(self, dtype=None, copy=False):
         array(['1999-12-31T23:00:00.000000000', '2000-01-01T23:00:00...'],
               dtype='datetime64[ns]')
         """
-        if (is_extension_array_dtype(self.dtype) or
-                is_datetime64tz_dtype(self.dtype)):
-            # TODO(DatetimeArray): remove the second clause.
-            # TODO(GH-24345): Avoid potential double copy
-            result = np.asarray(self._values, dtype=dtype)
-        else:
-            result = self._values
-
+        if is_datetime64tz_dtype(self.dtype) and dtype is None:
```

Comment:
This special case will be going away very soon. The problem is that `DatetimeArray.__array__` currently converts to UTC and returns an ndarray[M8[ns]]. We want it to return an ndarray[object] of Timestamps with the correct tz. That change breaks a few things elsewhere, so doing it as a followup.